src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -39,13 +39,15 @@ public TypeAssignment(ITypeReference reference, IEnumerable<Diagnostic> diagnost
         private readonly TypeManager typeManager;
         private readonly IBinder binder;
         private readonly IDictionary<SyntaxBase, TypeAssignment> assignedTypes;
+        private readonly IDictionary<FunctionSymbol, FunctionOverload> matchedFunctions;
```

Comment:
@majastrz This is new - an attempt to get which function was used to resolve the type

Reply:
Do we even need to cache these? I think the highest number of overloads we have for a function is like 4, so computing it every time we need it shouldn't be too bad.

Because of the `TryAdd()` comment below, lookups might fail against this dictionary, so what's the point?

Reply:
I didn't want do duplicate code responsible for matching function overload into the different visitor and avoid case where potentially different functions are matched.

Reply:
not valid anymore, functionality removed.